gapis/api/vulkan/transform_mapping_exporter.go

```diff
@@ -105,9 +109,50 @@ func (mappingTransform *mappingExporter) TransformCommand(ctx context.Context, i
 		mappingTransform.thread = inputCommands[0].Thread()
 	}
 
+	for _, cmd := range inputCommands {
+		if queueSubmit, ok := cmd.(*VkQueueSubmit); ok {
+			if err := mappingTransform.recordSubmittedRenderPass(ctx, queueSubmit, id.GetID(), inputState); err != nil {
+				return nil, err
+			}
+		}
+	}
+
 	return inputCommands, nil
 }
 
+func (mappingTransform *mappingExporter) recordSubmittedRenderPass(ctx context.Context, cmd *VkQueueSubmit, cmdID api.CmdID, inputState *api.GlobalState) error {
+	cmd.Extras().Observations().ApplyReads(inputState.Memory.ApplicationPool())
+	layout := inputState.MemoryLayout
+	stateObj := GetState(inputState)
+
+	submitInfos, err := cmd.PSubmits().Slice(0, uint64(cmd.SubmitCount()), layout).Read(ctx, cmd, inputState, nil)
+	if err != nil {
+		return err
+	}
+
+	for _, submitInfo := range submitInfos {
+		commandBuffers, err := submitInfo.PCommandBuffers().Slice(0, uint64(submitInfo.CommandBufferCount()), layout).Read(ctx, cmd, inputState, nil)
+		if err != nil {
+			return err
+		}
+
+		for _, cmdBuffer := range commandBuffers {
+			cmdBufferObj := GetState(inputState).CommandBuffers().Get(cmdBuffer)
+			for cmdIndex := 0; cmdIndex < cmdBufferObj.CommandReferences().Len(); cmdIndex++ {
+				currentCmd := cmdBufferObj.CommandReferences().Get(uint32(cmdIndex))
+				args := GetCommandArgs(ctx, currentCmd, stateObj)
+				if beginRenderPassArgs, ok := args.(VkCmdBeginRenderPassArgsʳ); ok {
+					if uint64(cmdID) > mappingTransform.numOfInitialCmds {
```

Comment:
I may be missing something here, but `cmdID` is an argument and `numOfInitialCmds` is not changed, i.e. this check could be done at the start of the function? I guess this should maybe be some other id?

Reply:
👍 good catch